docs/source/federation/implementing.md

```diff
@@ -151,36 +151,35 @@ On startup, the gateway will fetch the service capabilities from the running ser
 
 For existing services, it's likely that you've already implemented some form of authentication to convert a request into a user, or require some information passed to the service through request headers. `@apollo/gateway` makes it easy to reuse the context feature of Apollo Server to customize what information is sent to underlying services. Let's see what it looks like to pass user information along from the gateway to its services:
 
-```javascript{9-18,23-32}
+```javascript
 const { ApolloServer } = require('apollo-server');
 const { ApolloGateway, RemoteGraphQLDataSource } = require('@apollo/gateway');
 
+// highlight-start
+class AuthenticatedDataSource extends RemoteGraphQLDataSource {
+  willSendRequest({ request, context }) {
+    // pass the user's id from the context to underlying services
+    // as a header called `user-id`
+    request.http.headers.set('user-id', context.userId);
+  }
+}
+// highlight-end
+
 const gateway = new ApolloGateway({
   serviceList: [
     { name: 'products', url: 'http://localhost:4001' },
     // other services
   ],
+  // highlight-start
```

Comment:
Cool! This is so much more manageable than line numbers!

Reply:
Indeed, counting and updating the line numbers on this repeatedly was getting under my skin. These are long examples!